src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -155,5 +161,73 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
     // redirect to a GET request
     response.sendRedirect("/chat/" + conversationTitle);
+
+    //Get mentions in an individual single message
+    List<Mention> mentionsList = getMentions(message);
+
+  }
+  public List<Mention> getMentions(Message message){
+    List<Mention> mentions = new ArrayList<>();
+    String content = message.getContent();
+
+    for (int i = 0; i < content.length(); i++) {
```

Comment:
be careful with your indexes here.  You check to verify that i < content.length() but you frequently add one to i in the loop. That's a recipe for exceptions -- for instance if your string is "I type at signs like this: @" will probably throw.

One thing you can do is check if content.charAt(i) == '@" && (i == 0 || content.charAt(i - 1) == ' ')

You can do this in two ways:
(1) use indexOf
(2) use a state variable "inMention" to indicate that you're in a mention text while iterating.

The second one is probably simple to do quickly, like this:

bool inMention = false;
int startOfMention = -1;
int endOfMention = -1;
for (int i = 0; i < content.length(); i++) {
  if (content.charAt(i) == '@') {
    if (inMention) {
     inMention = false;
     break;
    } else if (i == 0 || content.charAt(i - 1) == ' ')) {
     inMention = true;
     startOfMention = i;
    }
  } else if (content.charAt(i) == ' ') {
    if (inMention) {
     endOfMention = i;
     mentions.add( ... somethingsomething startOfMention, endOfMention ...);
     inMention = false;
   }
 }
}

The first one would be like this:
for (int start = content.indexOf("@"); start >= 0 && start < content.length(); ) {
  if (start == 0 || content.charAt(start - 1) == " ") {
    int end = content.indexOf(" ", start);
    int atEnd = content.indexOf("@", start);
    if (atEnd >= 0 && atEnd < end) {
      // checking for @foo@bar instead of @foo @bar
      start = atEnd;
      break;
    }
... stuff ...not going to write it all out...
  }
  start = content.indexOf("@", start);
}